src/cljc/matthiasn/systems_toolbox/component.cljc

```diff
@@ -33,6 +37,16 @@
    :msgs-on-firehose      true
    :reload-cmp            true})
 
+(defn add-to-msg-seq
+  "Function for adding the current component ID to the sequence that the message has traversed
+  thus far. Component IDs can be added either when the message enters or leaves a component.
+  There's a test to avoid conjoining the same ID twice in a row."
+  [msg-meta cmp-id]
+  (let [cmp-seq (vec (:cmp-seq msg-meta))]
```

Comment:
I think it might be useful to only record new components when they enter. And then allow duplicates. I.e. if I misconfigure something, it'd be good to see that my message is trapped in a loop for instance.